src/Compilation.zig

```diff
@@ -188,6 +188,9 @@ libunwind_static_lib: ?CRTFile = null,
 /// Populated when we build the TSAN static library. A Job to build this is placed in the queue
 /// and resolved before calling linker.flush().
 tsan_static_lib: ?CRTFile = null,
+/// Populated when we build the TSAN dynamic library. A Job to build this is placed in the queue
+/// and resolved before calling linker.flush().
+tsan_dynamic_lib: ?CRTFile = null,
```

Comment:
I recommend to, instead of adding this new field, rename the other one from `tsan_static_lib` to `tsan_lib`.